src/AppBundle/Controller/MainController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use AppBundle\Entity\Todo;
+use Symfony\Component\HttpFoundation\Request;
+
+class MainController extends Controller
+{
+    /**
+     * @Route("/main", name="main"))
+     */
+    public function mainAction()
+    {   
+        if (!($this->container->get('security.authorization_checker')->isGranted('ROLE_USER'))) {
+            return $this->redirectToRoute('homepage');
+        }
+        $user = $this->container->get('security.token_storage')->getToken()->getUser();
+
+        $todos = $user->getTodos();
+
+        return $this->render('main.html.twig', array(
+            'todos' => $todos
+        ));
+    }
+
+
+    /**
+     * @Route("/main/create", name="create_todo"))
+     */
+    public function createAction(Request $request)
+    {   $user = $this->container->get('security.token_storage')->getToken()->getUser();
```

Comment:
"I should automatically check for unauthenticated users in the security.yml": YES
" noticing that I haven't checked for user permissions in createAction": ALSO YES

The issue here is that anyone can hit this endpoint without having to be a user. mainAction checks user role to figure out if it should show the main page or not, and createAction should also check role to figure out if the user is authenticated and can create a todo. Security is important